arch/arc/include/kernel_arch_func.h

```diff
@@ -31,9 +31,24 @@ extern "C" {
 #include <v2/irq.h>
 #endif
 
+static ALWAYS_INLINE _cpu_t *z_arch_curr_cpu(void)
+{
+#ifdef CONFIG_SMP
+	u32_t core;
+
+	core = z_arc_v2_core_id();
+
+	return &_kernel.cpus[core];
+#else
+	return &_kernel.cpus[0];
+#endif
+}
+
 static ALWAYS_INLINE void kernel_arch_init(void)
 {
 	z_irq_setup();
+	_current_cpu->irq_stack =
+		Z_THREAD_STACK_BUFFER(_interrupt_stack) + CONFIG_ISR_STACK_SIZE;
```

Comment:
single '_interrupt_stack' used by all CPUs? See earlier remarks, some additional comments or general overview on the SMP interrupt/stack design would help understanding this.

Reply:
kernel_arch_init is only called by master core, slave core will not call this.